config/config.go

```diff
@@ -749,6 +755,7 @@ var (
 )
 
 // Validate returns true if the config is valid. Ensures each key is valid and meets the required constraints.
+// Updates ValidatedKeySet once validated.
```

Comment:
I think this documentation could use more.
* Where does `c.JSONKeySet` come from?
  * If it came from JSON, why are we marshaling and using some `jwks.ParseKeySet` function? How is `ParseKeySet` different from simply unmarshalling?
* What's a key? The key of a Key-Value map object? An encryption key?

Looking at the library here, this goes pretty deep. I don't want to rope you into documenting raison d'etre for the whole function as part of this PR. I think for this abstraction level, we can get what we need by surfacing at the function level documentation the expected shape of a `JSONKeySet` for this call to succeed. Because `JSONKeySet` is just an "attribute map", we've lost the ability to navigate go structures to reveal what a good keyset looks like.

Reply:
updated with a basic example, and linked to what a JSONKeySet is derived from. I think the example + link should be enough to answer the questions, but let me know if it should go more in depth